Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +184,45 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)validateEntry:(id)maybeFileURL
+{
+    NSString *errorMessage = nil;
+
+    auto* url = dynamic_objc_cast<NSURL>(maybeFileURL);
+    if (!url)
+        errorMessage = @"The NSDictionary associated with _WKReadAccessFileURLsOption may only contain NSURL objects.";
+    else if (!url.fileURL)
+        errorMessage = @"_WKReadAccessFileURLsOption requires its NSURL objects to be file URLs.";
+
+    if (!errorMessage)
+        return;
+
+    if (self.readOnlyAccessPaths.get().count) {
+        [self.readOnlyAccessPaths removeAllObjects];
+        [self clearConfiguration];
+    }
+    [NSException raise:NSInvalidArgumentException format:@"%@", errorMessage];
+}
+
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessFileURLs = dynamic_objc_cast<NSArray<NSURL *>>(options[_WKReadAccessFileURLsOption]);
+    if (readAccessFileURLs.count > maximumReadOnlyAccessPaths)
+        [NSException raise:NSInvalidArgumentException format:@"_WKReadAccessFileURLsOption may have at most two additional directories."];
```

Comment:
I think we went to do the same thing here as where we raise the exception above in `validateEntry:` so they should share code.

Reply:
The same thing being the "remove any items that are in the array from a previous call" code.

Reply:
Will do.